js/stats.js

```diff
@@ -1,173 +1,222 @@
-
-  console.log("sorting by greedy");
-  console.time('time taken (greedy)');
-  var sorted1 = sSplitHelper(numGroups, numPictures, targetRating, pictures, sSplitGreedy);
-  console.timeEnd('time taken (greedy)');
-  printStats(sorted1, targetRating);
-
-  console.log();
-
-  console.log("sorting by round robin");
-  console.time('time taken (round robin)');
-  var sorted2 = sSplitHelper(numGroups, numPictures, targetRating, pictures, sSplitRR);
-  console.timeEnd('time taken (round robin)');
-    printStats(sorted2, targetRating);
-
-  console.log();
-
-  console.log("sorting by random");
-  console.time('time taken (random)');
-  var sorted3 = sSplitHelper(numGroups, numPictures, targetRating, pictures, sSplitRandom);
-  console.timeEnd('time taken (random)');
-    printStats(sorted3, targetRating);
-}
-
-
-/* update this function with parameters */
-testSSplit(4, 10, 8, 100, 10);
-
-      
+/*
+    Author: Tony J Huang
+    Group: Team Stimulus
+    Created At: 2/18/15
+
+    This is the stats module, affectionately and globally referred 
+    to as _$_. This module provides the solutions to all of your 
+    statistical needs, including standard deviations, means, medians, 
+    modes, and much much more (coming soon [tm])!
+
+    Check the README file for complete documentation on public methods.
+
+    Note: This module relies heavily on Underscore.js 
+        (http://underscorejs.org/). If you don't have this library 
+        installed then you're out of luck my friend.
+
+    Note: Unless otherwise noted, all code henceforth in this file
+        has been authored by me, Tony J Huang. 
+*/
+
+/*
+    Global export for _$_ variable. 
+
+    Sample usages include: 
+      _$_.sSplit(...);
+      _$_.stDev(...);
+*/
+var _$_ = (function() {
+    var my = {};
+
+
+    // Given a object, checks if it has a value at fieldName.
+    // Otherwise throws an error.
+    //
+    // Creation date: 3/1/15
+    // Modifications list:
+    //
+    function ensure(params, fieldName) {
+        if(params[fieldName])
+            return params[fieldName];
+        throw new Error("missing param field: " + fieldName);
+    }
+
+
+    // Validates the input as a properly formatted Array of Pictures.
+    //
+    // Creation date: 3/1/15
+    // Modifications list:
+    //
+    function validatePictures(objects) {
+        if(!_.isArray(pictures))
+            throw new Error(
+                "Pictures argument is not an Array of Picture");
+
+        if(!_.every(objects, o => o instanceof Picture))
+            throw new Error(
+                "Pictures argument is not properly formatted, ie: " + o);
+    }
+
+
+    // Validates the other user defined input params ansd makes sure
+    // they make sense/are possible.
+    //
+    // Creation date: 3/1/15
+    // Modifications list:
+    //
+    function validateNumArgs(numGroups, numPictures, pictures) {
```

Comment:
What about if we have a number of pictures that can't be divided evenly into the number of groups?